File: lib/go/internal/schema/GetPathDocumentYAMLCoordinatesPseudoJSON.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package schema

func CreatePathDocumentYAMLCoordinatesPseudoJSONLocator(text string) (DocumentLocator, error) {
	return createDocumentLocatorFromYAMLText(text)
}

func GetPathDocumentYAMLCoordinatesPseudoJSON(path []any, document string) (location map[string]any) {
	locator, err := CreatePathDocumentYAMLCoordinatesPseudoJSONLocator(document)
	if err != nil {
		return defaultDocumentCoordinates()
	}
	return locator(path)
}

File: lib/ts/src/internal/schema/GetPathDocumentYamlCoordinatesPseudoJson.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { createDocumentLocatorFromYamlText } from './BuildDocumentLocatorFromYamlAst.js';
import { Coordinates, DocumentLocator, Path } from './DocumentLocators.js';

export function createPathDocumentYamlCoordinatesPseudoJsonLocator(text: string): DocumentLocator {
    return createDocumentLocatorFromYamlText(text);
}

export function getPathDocumentYamlCoordinatesPseudoJson(path: Path, document: string): Coordinates {
    return createPathDocumentYamlCoordinatesPseudoJsonLocator(document)(path);
}
